contract_compassion/models/contracts.py

```diff
@@ -215,6 +218,18 @@ def _compute_contract_duration(self):
                 contract.contract_duration = (
                     end_date - contract_start_date).days
 
+    def _compute_last_letter(self):
+        """
+        Get the date of the last letter sent to the child by the sponsor and
+        compute since how many days he didn't write him any letter
+        :return: int value
+        """
+        self.last_letter = (date.today() - datetime.strptime(
+            self.env['correspondence'].search([(
+                'direction', '=', 'Supporter To Beneficiary'), (
+                'sponsorship_id', '=', self.id)], limit=1).scanned_date,
```

Comment:
Did you think of the case if there are no letters? It should not fail but return the number of days since the creation of the sponsorship (using the field create_date)